### PR TITLE
Treat frustum changes as camera changes in Scene#render

### LINKED ISSUE #6812: requestRenderMode Bug on resize
I believe I've found a bug related to requestRenderMode in Firefox. I asked about it in the forums (https://groups.google.com/forum/#!topic/cesium-dev/cHR1qyjMKeo) and a community member suggested I file a bug.

Essentially, Cesium doesn't stop the update/render process in Firefox until you interact (move, zoom) with the globe after the page initially loads. You can see this happen by visiting the Sandcastle demo at https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/?src=Scene%20Rendering%20Performance.html. In Chrome, the FPS flatline to "N/A" within a few seconds after page load. The FPS never flatline after the page load in Firefox, however, until you move the globe or zoom in.

Cesium: 1.4.7
Browser: Firefox 61.0.1
OS: CentOS 7.5

### PR DESCRIPTION
**Pull request by lukesanantonio**
Before this change, the user could update the frustum but not the camera by resizing the viewer. The Scene would continue to render even when `requestRenderMode` was enabled because it recognized the frustum had changed, but would not update the cached camera to mark the change as handled. This commit changes `cameraEqual` (in `Scene.js`) to take into account the camera frustum.

Also adds `equalsEpsilon` methods to the `*Frustum` types to make this work.

Fixes #6812

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @lukesanantonio!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/frustum-render-fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by mramato** [comment]
Thanks for tracking this down @lukesanantonio!  @ggetz can you review?  @bagnell are you okay with adding these methods to Frustum?

> Fixes #6812

Mention this explicitly in CHANGES and link to this PR or the original issue.  Look at other recent bullets in the markdown for examples

File: Source/Scene/Scene.js [hunk 1]
```diff
@@ -3310,10 +3315,8 @@ define([
         tryAndCatchError(this, time, update);
         this._postUpdate.raiseEvent(this, time);
 
-        this._frustumChanged = !this._camera.frustum.equals(this._cameraClone.frustum);
-
         var cameraChanged = checkForCameraUpdates(this);
-        var shouldRender = !this.requestRenderMode || this._renderRequested || cameraChanged || this._frustumChanged || this._logDepthBufferDirty || (this.mode === SceneMode.MORPHING);
```
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Scene/Scene.js]: It's used on Line 1513. It didn't look like cameraChanged (or something like it) could be a drop-in replacement so I didn't change it.

File: Source/Core/OrthographicFrustum.js [hunk 2]
```diff
@@ -273,5 +275,30 @@ define([
...
+        }
+
+        update(this);
+        update(other);
+
+        return (CesiumMath.equalsEpsilon(this.width, other.width, relativeEpsilon, absoluteEpsilon) &&
+                CesiumMath.equalsEpsilon(this.aspectRatio, other.aspectRatio, relativeEpsilon, absoluteEpsilon) &&
+                CesiumMath.equalsEpsilon(this.near, other.near, relativeEpsilon, absoluteEpsilon) &&
+                CesiumMath.equalsEpsilon(this.far, other.far, relativeEpsilon, absoluteEpsilon) &&
+                this._offCenterFrustum.equalsEpsilon(other._offCenterFrustum, relativeEpsilon, absoluteEpsilon));
```
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Core/OrthographicFrustum.js]: It seems like the `*OffCenter*` variants don't check `fov`,`width`, or `aspectRatio`. Is that still alright?

[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Core/OrthographicFrustum.js]: Oh I didn't realize you only meant the ones you highlighted. `near` and `far` checks are def redundant

File: Source/Core/OrthographicFrustum.js [hunk 3]
```diff
@@ -273,5 +275,30 @@ define([
                 this._offCenterFrustum.equals(other._offCenterFrustum));
     };
 
+    /**
+     * Compares the provided OrthographicFrustum componentwise and returns
+     * <code>true</code> if they pass an absolute or relative tolerance test,
+     * <code>false</code> otherwise.
+     *
+     * @param {OrthographicFrustum} [other] The right hand side OrthographicFrustum.
```
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Core/OrthographicFrustum.js]: That's fair, is there a way to mark this in the docs or should I just remove the `defined(other)` part of the condition?

File: Source/Scene/Scene.js [hunk 4]
```diff
@@ -1488,7 +1488,8 @@ define([
                Cartesian3.equalsEpsilon(camera0.direction, camera1.direction, epsilon) &&
                Cartesian3.equalsEpsilon(camera0.up, camera1.up, epsilon) &&
                Cartesian3.equalsEpsilon(camera0.right, camera1.right, epsilon) &&
-               Matrix4.equalsEpsilon(camera0.transform, camera1.transform, epsilon);
+               Matrix4.equalsEpsilon(camera0.transform, camera1.transform, epsilon) &&
+               camera0.frustum.equalsEpsilon(camera1.frustum, epsilon);
```
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Scene/Scene.js]: I'm having trouble getting the test to fail on master. I'll try again tomorrow but do you have any tips / ideas?

**Comment by lukesanantonio** [comment]
Sorry for the delay, made some changes but I still need some help on a few things @ggetz @hpinkos

File: Source/Core/OrthographicFrustum.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ggetz) [reply Source/Core/OrthographicFrustum.js]: Just update the docs by removing the brackets, `[]`.

File: Source/Scene/Scene.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ggetz) [reply Source/Scene/Scene.js]: Take a look at the other `requestRender` tests, if you haven't already.

File: Source/Scene/Scene.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggetz) [reply Source/Scene/Scene.js]: Yes, that's fine to leave.

File: Source/Scene/Scene.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Scene/Scene.js]: Got it, the newest commit has a test that failed before and passes with these changes.

File: Source/Scene/Scene.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Scene/Scene.js]: Gotcha

File: Source/Core/OrthographicFrustum.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukesanantonio) [reply Source/Core/OrthographicFrustum.js]: Ha! I didn't realize those brackets mark them optional. Will fix

**Comment by lukesanantonio** [comment]
Okay I think this is all good @ggetz

**Comment by ggetz** [comment]
Looks good, thanks @lukesanantonio !

**Pull request merged**